### PR TITLE
Bug fixes related to Members

### PR DESCRIPTION
**Pull request by richa-copods**
Fixed the following issues:

- While creating test, if description is entered and then removed then user is able to click on Next, but ideally it should not be
- Show error messages while adding member
- When section’s page is empty, it should not show side section
- Fixed the variable names' issue in cypress test cases

### CODE REVIEW

File: app/routes/members.tsx [hunk 1]
```diff
@@ -55,25 +52,25 @@ export const action: ActionFunction = async ({ request }) => {
 
     if (typeof firstName !== 'string' || firstName.length === 0) {
       return json<ActionData>(
-        { errors: { firstName: 'firstName is required' } },
+        { errors: { title: 'FirstName is required', status: 400 } },
```
Comment (anuragP04) [inline app/routes/members.tsx:55]: we can use constants for these errors(for all error or success statements)

File: app/routes/members.tsx [hunk 2]
```diff
@@ -104,49 +110,55 @@ export const action: ActionFunction = async ({ request }) => {
...
+        { errors: { title: 'Description is required', status: 400 } },
         { status: 400 }
       )
     }
     let deleteHandle = null
     await deleteUserById(formData.get('id') as string)
       .then((res) => {
         deleteHandle = json<ActionData>(
-          { resp: { status: 'Deleted Successfully..!' } },
+          { resp: { title: 'Deleted Successfully..!', status: 200 } },
```
Comment (anuragP04) [inline app/routes/members.tsx:121]: use constants for these

File: app/routes/sections.tsx [hunk 3]
```diff
@@ -138,6 +151,7 @@ export default function SectionPage() {
 
   useEffect(() => {
     if (data.sections.length && !data.selectedSectionId) {
+      console.log('1', data?.filters)
```
Comment (anuragP04) [inline app/routes/sections.tsx:154]: remove console.log

File: app/routes/sections.tsx [hunk 4]
```diff
@@ -152,6 +166,7 @@ export default function SectionPage() {
           [sortBy]: order,
         },
       }
+      console.log('2', data?.filters)
```
Comment (anuragP04) [inline app/routes/sections.tsx:169]: remove console.log

File: app/routes/sections.tsx [hunk 5]
```diff
@@ -165,7 +180,8 @@ export default function SectionPage() {
     if (action) {
       if (action.resp?.status === 200) {
         setShowAddSectionModal(false)
-        toast.success('Section added successfully..!')
+        toast.success(action?.resp?.title)
+        console.log('3', data?.filters)
```
Comment (anuragP04) [inline app/routes/sections.tsx:184]: remove console.log

File: app/routes/sections.tsx [hunk 6]
```diff
@@ -193,51 +209,58 @@ export default function SectionPage() {
...
-            />
-          </div>
+        {data.sections.length > 0 ? (
+          <div
+            className={`flex flex-1 overflow-hidden ${
+              sectionDetailFull ? '' : 'gap-12'
+            }`}
+          >
+            {/* section list */}
+            <div className={`${sectionDetailFull ? 'hidden' : ''}`}>
```
Comment (anuragP04) [inline app/routes/sections.tsx:219]: why you are assigning '' if there is nothing to assign? we can use && for this.

File: app/routes/sections.tsx [hunk 7]
```diff
@@ -193,51 +209,58 @@ export default function SectionPage() {
...
-              ) : (
-                <Icon
-                  icon={'akar-icons:circle-chevron-left-fill'}
-                  className="cursor-pointer text-4xl text-primary"
-                />
-              )}
-            </span>
-            <Outlet />
+            {/* section details */}
+            <div className={`z-10 flex flex-1 items-center `}>
```
Comment (anuragP04) [inline app/routes/sections.tsx:234]: remove extra space

File: app/components/assessment/Question.tsx [hunk 8]
```diff
@@ -34,7 +34,7 @@ const Question = () => {
 
   return (
     <form method="post" className="flex h-full flex-col gap-9">
-      <div className="flex h-full max-h-full flex-1 flex-1 gap-9 overflow-auto">
+      <div className="flex h-full  max-h-full flex-1 gap-9 overflow-auto">
```
Comment (DhruvCopods) [inline app/components/assessment/Question.tsx:37]: remove extra space

File: app/components/members/MembersHeader.tsx [hunk 9]
```diff
@@ -5,32 +5,32 @@ import { useLoaderData } from '@remix-run/react'
...
+
   const [open, setOpen] = useState(false)
+
   useEffect(() => {
-    err ? setOpen(true) : setOpen(false)
-  }, [actionStatus, err])
-  const membersData = useLoaderData()
+    if (actionStatus) {
+      setOpen(false)
+    }
```
Comment (DhruvCopods) [inline app/components/members/MembersHeader.tsx:18]: can be a single line ( if () setOpen() )

File: app/components/sections/Sections.tsx [hunk 10]
```diff
@@ -3,7 +3,6 @@ import type { Section } from '~/interface/Interface'
 import { useResolvedPath, useLocation, NavLink } from '@remix-run/react'
 import {} from '@remix-run/react'
```
Comment (DhruvCopods) [inline app/components/sections/Sections.tsx:4]: remove since importing nothing

File: app/components/sections/Sections.tsx [hunk 11]
```diff
@@ -3,7 +3,6 @@ import type { Section } from '~/interface/Interface'
...
 import SortFilter from '../SortFilter'
-import { sectionsConstants } from '~/constants/common.constants'
 
 const SectionLink = ({ section }: { section: any }) => {
```
Comment (DhruvCopods) [inline app/components/sections/Sections.tsx:7]: type of any give to section, change that

**Review by anuragP04 — APPROVED** [review]

**Pull request closed without merging**